tests/queries/0_stateless/03164_parallel_replicas_range_filter_min_max.sql

```diff
@@ -0,0 +1,57 @@
+DROP TABLE IF EXISTS range_filter_custom_range_test;
+
+CREATE TABLE range_filter_custom_range_test (k Int64) ENGINE=MergeTree ORDER BY k;
+
+INSERT INTO range_filter_custom_range_test SELECT number + 5 from numbers(10);
+
+SELECT count() FROM 
+(SELECT *
+FROM cluster(test_cluster_one_shard_three_replicas_localhost, currentDatabase(), range_filter_custom_range_test)
+SETTINGS prefer_localhost_replica = 0, max_parallel_replicas = 3, distributed_group_by_no_merge=0,
+parallel_replicas_custom_key = 'k', parallel_replicas_custom_key_filter_type = 'range', 
+parallel_replicas_custom_key_range_lower = 5, parallel_replicas_custom_key_range_upper=15);
+
+SELECT count() FROM 
+(SELECT *
+FROM cluster(test_cluster_one_shard_three_replicas_localhost, currentDatabase(), range_filter_custom_range_test)
+SETTINGS prefer_localhost_replica = 0, max_parallel_replicas = 3, distributed_group_by_no_merge=0,
+parallel_replicas_custom_key = 'k', parallel_replicas_custom_key_filter_type = 'range', 
+parallel_replicas_custom_key_range_lower = 4, parallel_replicas_custom_key_range_upper=14);
+
+SELECT count() FROM 
+(SELECT *
+FROM cluster(test_cluster_one_shard_three_replicas_localhost, currentDatabase(), range_filter_custom_range_test)
+SETTINGS prefer_localhost_replica = 0, max_parallel_replicas = 3, distributed_group_by_no_merge=0,
+parallel_replicas_custom_key = 'k', parallel_replicas_custom_key_filter_type = 'range', 
+parallel_replicas_custom_key_range_lower = 6, parallel_replicas_custom_key_range_upper=17);
+
+
+SELECT count() FROM 
+(SELECT *
+FROM cluster(test_cluster_one_shard_three_replicas_localhost, currentDatabase(), range_filter_custom_range_test)
+SETTINGS prefer_localhost_replica = 0, max_parallel_replicas = 3, distributed_group_by_no_merge=0,
+parallel_replicas_custom_key = 'k', parallel_replicas_custom_key_filter_type = 'range', 
+parallel_replicas_custom_key_range_lower = 0, parallel_replicas_custom_key_range_upper=15);
+
+SELECT count() FROM 
+(SELECT *
+FROM cluster(test_cluster_one_shard_three_replicas_localhost, currentDatabase(), range_filter_custom_range_test)
+SETTINGS prefer_localhost_replica = 0, max_parallel_replicas = 3, distributed_group_by_no_merge=0,
+parallel_replicas_custom_key = 'k', parallel_replicas_custom_key_filter_type = 'range', 
+parallel_replicas_custom_key_range_lower = 15, parallel_replicas_custom_key_range_upper=25);
```

Comment:
I'm not sure I follow how do we get 10 values if we filter values in range `[15, 25]`?
And we store values `[5, 14]`

Reply:
I did not want to change the behaviour of the actual query with these settings. So no matter the value of the `parallel_replicas_custom_key_range_upper` and `parallel_replicas_custom_key_range_lower` `SELECT count()` will still return all values. I think this behaviour makes the most sense; these settings DO NOT change the query logic, rather they change how it is parallelized over replicas. If the user wants to change what is returned by the query they can add a WHERE clause. Please let me know if you disagree with this reasoning.